systemservices/marketplace/src/contracts/version.ts

```diff
@@ -10,33 +10,70 @@ const getServiceVersions = async (contract: Marketplace, sid: string): Promise<V
-    versionsPromise.push(getServiceVersionWithIndex(contract, sid, j))
+    try {
+      const version = normalizeVersion(await getServiceVersionWithIndex(contract, sid, j))
+      if (validVersion(version)) {
+        versions.push(version)
+      }
+    } catch(e) {
+      console.warn(e.toString())
+    }
   }
-  const versions = await Promise.all(versionsPromise)
-  return versions.filter(version => version !== undefined) as Version[]
+  return versions
 }
 
-const getServiceVersionWithIndex = async (contract: Marketplace, sid: string, versionIndex: BigNumber): Promise<Version|undefined> => {
+const getServiceVersionWithIndex = async (contract: Marketplace, sid: string, versionIndex: BigNumber): Promise<Version> => {
   const versionHash = hexToHash(await contract.methods.serviceVersionHash(asciiToHex(sid), versionIndex.toString()).call())
   return getServiceVersion(contract, versionHash)
 }
 
-const getServiceVersion = async (contract: Marketplace, versionHash: string): Promise<Version|undefined> => {
+const getServiceVersion = async (contract: Marketplace, versionHash: string): Promise<Version> => {
   const versionHashHex = hashToHex(versionHash)
   if (!await contract.methods.isServiceVersionExist(versionHashHex).call()) {
-    return
+    throw new Error('Service does not exists')
   }
   const version = await contract.methods.serviceVersion(versionHashHex).call()
-  const manifestData = await getManifest(hexToAscii(version.manifestProtocol), hexToAscii(version.manifest))
+  const manifest = await getManifest(hexToAscii(version.manifestProtocol), hexToAscii(version.manifest))
+  if (!manifest) {
+    throw new Error('Unable to get the manifest')
+  }
   return {
     versionHash: versionHash,
-    manifest: hexToAscii(version.manifest),
-    manifestProtocol: hexToAscii(version.manifestProtocol),
-    manifestData: manifestData,
+    manifest: manifest,
     createTime: parseTimestamp(version.createTime),
   }
 }
 
+const validVersion = (version: Version): boolean => {
+  return version.versionHash != '' &&
+    !!version.manifest &&
+    !!version.manifest.service &&
+    !!version.manifest.service.hash &&
+    !!version.manifest.service.hashVersion &&
+    !!version.manifest.service.definition &&
+    !!version.manifest.service.deployment
+}
+
+const normalizeVersion = (version: any): Version => {
+  return {
+    createTime: version.createTime,
+    versionHash: version.versionHash,
+    manifest: {
+      version: version.manifest.version.toString(),
+      service: {
+        definition: version.manifest.service.definition,
+        deployment: {
+          type: version.manifest.service.deployment.type || 'ipfs',
+          source: version.manifest.service.deployment.source,
+        },
+        hash: version.manifest.service.hash,
+        hashVersion: version.manifest.service.hashVersion,
+        readme: version.manifest.service.readme || '',
+      }
+    }
+  }
+}
+
```

Comment:
I think `validVersion` and `normalizeVersion` should be removed in favor of validation done in https://github.com/mesg-foundation/core/pull/829.

Reply:
I can remove but then it will break some stuff because now the core is actually validating some data (because we don't have an any anymore but the actual structure of data) and this breaks a lot of services.

I was thinking more to move the validation in this function `validate` but i'm ok to revert as long as the other one is merged pretty much at the same time in order not to break everything